pkg/runtest/run.go

```diff
@@ -305,13 +306,10 @@ func parseProg(target *prog.Target, dir, filename string, requires map[string]bo
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("failed to read %v: %w", filename, err)
 	}
-	properties := parseRequires(data)
-	// Need to check arch requirement early as some programs
-	// may fail to deserialize on some arches due to missing syscalls.
-	if !checkArch(properties, target.Arch) || !match(properties, requires) {
-		return nil, nil, nil, nil
+	p, properties, err := manager.ParseSeedWithRequirements(target, data, requires)
+	if errors.Is(err, manager.ErrSkippedTest) {
+		return nil, properties, nil, nil
```

Comment:
Nit: do you need to return `properties` in this case or can just return `nil`?

Reply:
Fixed.